services/frontend-service/src/ui/utils/Links.tsx

```diff
@@ -146,3 +147,28 @@ export const DisplayManifestLink: React.FC<{ displayString: string; app: string;
     }
     return null;
 };
+
+export const ProductVersionLink: React.FC<{ env: string; groupName: string }> = (props): JSX.Element | null => {
+    const { env, groupName } = props;
+    const location = useLocation();
+    const separator = groupName === '' ? '' : '%2F';
+    var queryParams = location?.search ?? '';
+    const currentLink = window.location.href;
+    if (queryParams === '') {
+        queryParams = '?env=' + groupName + separator + env;
+    } else if (queryParams.includes('env=')) {
+        const start = queryParams.indexOf('env=');
+        const end = queryParams.indexOf('&', start) === -1 ? queryParams.length : queryParams.indexOf('&', start);
+        queryParams = queryParams.replace(queryParams.substring(start, end), 'env=' + groupName + separator + env);
+    } else {
+        queryParams += '&env=' + groupName + separator + env;
+    }
+    const addParam = currentLink.split('?');
```

Comment:
please restructure this.
E.g use a function similar to this one https://stackoverflow.com/questions/1714786/query-string-encoding-of-a-javascript-object
1) create object with desired query keys and values
2) call function
3) add result to the link.